pom.xml

```diff
@@ -131,6 +132,18 @@
             <groupId>com.microsoft.azure.kusto</groupId>
             <artifactId>kusto-ingest</artifactId>
             <version>${kusto-sdk.version}</version>
+            <exclusions>
```

Comment:
@tanmaya-panda1  , can you attach the tests that you had done for all formats, just to be sure that all of these work.

- Avro
- JSON
- ORC
- String

Reply:
**CSV**

value.converter : org.apache.kafka.connect.storage.StringConverter

mapping :

"kusto.tables.topics.mapping": "[{'topic': 'storm-events','db': 'Storms', 'table': 'Storms','format': 'csv'}]"

table creation in ADX for CSV:

.create table Storms ingestion csv mapping 'Storms_CSV_Mapping' '[{"Name":"StartTime","datatype":"datetime","Ordinal":0}, {"Name":"EndTime","datatype":"datetime","Ordinal":1},{"Name":"EventId","datatype":"int","Ordinal":2},{"Name":"State","datatype":"string","Ordinal":3},{"Name":"EventType","datatype":"string","Ordinal":4},{"Name":"Source","datatype":"string","Ordinal":5}]'


**JSON**

value.converter : org.apache.kafka.connect.storage.StringConverter

"kusto.tables.topics.mapping": "[{'topic': 'storm-events','db': 'Storms', 'table': 'Storms','format': 'json'}]"

table creation in ADX for JSON:

.create table Storms ingestion json mapping 'Storms_Json_Mapping' '[{ "column" : "StartTime", "datatype" : "datetime", "path" : "$.StartTime"},{ "column" : "EndTime", "datatype" : "datetime" ,"path" : "$.EndTime" }, { "column" : "EventId", "datatype" : "int" ,"path" : "$.EventId" }, { "column" : "State", "datatype" : "string" ,"path" : "$.State" },{ "column" : "EventType", "datatype" : "string" ,"path" : "$.EventType" },{ "column" : "Source", "datatype" : "string" ,"path" : "$.Source" }]'


**Avro **

value.converter : io.confluent.connect.avro.AvroConverter

mapping : 

"kusto.tables.topics.mapping": "[{'topic': 'storm-events','db': 'Storms', 'table': 'Storms','format': 'avro'}]"

avro schema definition :

{
  "type": "record",
  "name": "schema",
  "namespace": "com.microsoft",
  "fields": [
    {
      "name": "StartTime",
      "type": "long",
      "doc": "start time"
    },
    {
      "name": "EndTime",
      "type": "long",
      "doc": "End Time"
    },
    {
      "name": "EventId",
      "type": "int",
      "doc": "Event Id"
    },
    {
      "name": "State",
      "type": "string",
      "doc": "State"
    },
    {
      "name": "EventType",
      "type": "string",
      "doc": "Event Type"
    },
    {
      "name": "Source",
      "type": "string",
      "doc": "Source"
    }
  ],
  "doc:": "A basic schema"
}

table creation in ADX for avro :

.create table MyTable ingestion avro mapping 'Storms_Mapping_AVRO'
'[
  {"column": "StartTime", "Properties": {"Field": "StartTime"}},
  {"column": "EndTime",      "Properties": {"Field": "EndTime"}},
  {"column": "EventId",      "Properties": {"Field": "EventId"}},
  {"column": "State",      "Properties": {"Field": "State"},
  {"column": "EventType",  "Properties": {"Field": "EventType"}},
  {"column": "Source",     "Properties": {"Field": "Source"}}
]'


**PARQUET**


value.converter : org.apache.kafka.connect.converters.ByteArrayConverter

mapping :

"kusto.tables.topics.mapping": "[{'topic': 'storm-events','db': 'Storms', 'table': 'Storms','format': 'parquet'}]"

table creation in ADX for PARQUET:

.create table MyTable ingestion parquet mapping 'Storms_Mapping_Parquet' 
'[
  {"Column": "StartTime", "Properties": {"Path": "$.StartTime"}},
  {"Column": "EndTime",      "Properties": {"Path": "$.EndTime"}},
  {"Column": "EventId",      "Properties": {"Path": "$.EventId"}},
  {"Column": "State",      "Properties": {"Path": "$.State"}},
  {"Column": "EventType",  "Properties": {"Path": "$.EventType"}},
  {"Column": "Source",     "Properties": {"Path": "$.Source"}}
]'




**ORC**

value.converter : org.apache.kafka.connect.converters.ByteArrayConverter

mapping :

"kusto.tables.topics.mapping": "[{'topic': 'storm-events','db': 'Storms', 'table': 'Storms','format': 'orc'}]"

table creation in ADX for ORC:

.create table MyTable ingestion orc mapping "Storms_Mapping_ORC"
[
  {"Column": "StartTime", "Properties": {"Path": "$.StartTime"}},
  {"Column": "EndTime",      "Properties": {"Path": "$.EndTime"}},
  {"Column": "EventId",      "Properties": {"Path": "$.EventId"}},
  {"Column": "State",      "Properties": {"Path": "$.State"}},
  {"Column": "EventType",  "Properties": {"Path": "$.EventType"}},
  {"Column": "Source",     "Properties": {"Path": "$.Source"}}
]

Reply:
Just for the sake of this discussion - we don't think there are really parquet and orc users - don't waste too much time on testing these